code/game/atoms.dm

```diff
@@ -1157,34 +1157,75 @@
   * Must return  parent proc ..() in the end if overridden
   */
 /atom/proc/tool_act(mob/living/user, obj/item/I, tool_type)
+	var/list/processing_recipes = list() //List of recipes that can be mutated by sending the signal
+	. |= SEND_SIGNAL(src, COMSIG_TOOL(tool_type), user, I, processing_recipes)
+	message_admins(COMSIG_TOOL(tool_type))
```

Comment:
debugging line

Reply:
no i want the admins to know whenever someone uses a tool bro